packages/rum/src/domain/deflate/deflateEncoder.spec.ts

```diff
@@ -20,28 +20,175 @@ describe('createDeflateEncoder', () => {
+      expect(finishCallbackSpy.calls.allArgs()).toEqual([
+        [
+          {
+            output: new Uint8Array([...ENCODED_FOO, ...TRAILER]),
+            outputBytesCount: 4,
+            rawBytesCount: 3,
+            encoding: 'deflate',
+          },
+        ],
+        [
+          {
+            output: new Uint8Array([...ENCODED_BAR, ...TRAILER]),
+            outputBytesCount: 4,
+            rawBytesCount: 3,
+            encoding: 'deflate',
+          },
+        ],
+      ])
+    })
+  })
+
+  describe('finishSync()', () => {
+    it('returns the encoded data up to this point and any pending data', () => {
+      const encoder = createDeflateEncoder(configuration, worker, DeflateEncoderStreamId.REPLAY)
+      encoder.write('foo')
+      encoder.write('bar')
+
+      worker.processNextMessage()
+
+      expect(encoder.finishSync()).toEqual({
+        output: new Uint8Array([...ENCODED_FOO, ...TRAILER]),
+        outputBytesCount: 4,
+        rawBytesCount: 3,
+        pendingData: 'bar',
+        encoding: 'deflate',
+      })
+    })
+
+    it('cancels pending write callbacks', () => {
+      const encoder = createDeflateEncoder(configuration, worker, DeflateEncoderStreamId.REPLAY)
+      const writeCallbackSpy = jasmine.createSpy()
+      encoder.write('foo', writeCallbackSpy)
+      encoder.write('bar', writeCallbackSpy)
+      encoder.finishSync()
+
+      worker.processAllMessages()
 
-    expect(writeCallbackSpy).toHaveBeenCalledTimes(2)
-    expect(encoder.encodedBytes).toEqual(new Uint8Array([...ENCODED_FOO, ...ENCODED_BAR, ...TRAILER]))
-    expect(encoder.encodedBytesCount).toBe(7)
-    expect(encoder.rawBytesCount).toBe(6)
+      expect(writeCallbackSpy).not.toHaveBeenCalled()
+    })
+
+    it('marks the encoder as empty', () => {
+      const encoder = createDeflateEncoder(configuration, worker, DeflateEncoderStreamId.REPLAY)
+      encoder.write('foo')
+      encoder.finishSync()
+      expect(encoder.isEmpty).toBe(true)
+    })
+
+    it('supports calling finishSync() while another finish() call is pending', () => {
+      const encoder = createDeflateEncoder(configuration, worker, DeflateEncoderStreamId.REPLAY)
+      const finishCallbackSpy = jasmine.createSpy<(result: EncoderResult<Uint8Array>) => void>()
+      encoder.write('foo')
+      encoder.finish(finishCallbackSpy)
+      encoder.write('bar')
+      expect(encoder.finishSync()).toEqual({
+        output: new Uint8Array(0),
+        outputBytesCount: 0,
+        rawBytesCount: 0,
+        pendingData: 'bar',
+        encoding: 'deflate',
+      })
+
+      worker.processAllMessages()
+
+      expect(finishCallbackSpy).toHaveBeenCalledTimes(1)
+      expect(finishCallbackSpy.calls.allArgs()).toEqual([
+        [
+          {
+            output: new Uint8Array([...ENCODED_FOO, ...TRAILER]),
+            outputBytesCount: 4,
+            rawBytesCount: 3,
+            encoding: 'deflate',
+          },
+        ],
+      ])
+    })
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏In this case, is not it risky to rely on the previous finish callback? 
What about:
- discarding the initial finish callback
- return with finishSync:
  - output: all compressed data so far 
  - pendingData: all pending data, regardless of if is has been added after a previous finish